### ISSUE TITLE
how to change/pass uniforms to a shader edited in the shader editor? 

### ISSUE THREAD

**Issue by shantee** [tentone/nunuStudio#332]
Hello

Even if it is still incomplete the shader editor is still nice to organize shaders. 
I would like to know how to pass uniforms with a script to a shader I have created in the shader editor (created by clicking materials > shadermaterial  in the ressources manager) 
Thanks

**Comment by tentone** [comment]
Hello

Thanks a lot for the question you can access your shader object by using the program.getMaterialByName('') method.

I am preparing a small tutorial with examples for booth ShaderMaterial and RawShaderMaterial.

Probably i should on a better solution for the shader editor.

**Comment by shantee** [comment]
Yes I actually already found the way to access my shader (while "playing" inside the dev console).
So I can get a shader created inside the "shader editor"  and  assign it to an object. But unfortunately I still can't figure out how to change its uniforms.
let's take a very simple shader having only "time" as uniforms

```
uniforms float time; 
[...]
```
I can display its data into the console 
`console.log(myShader)  `
but the uniforms part is always empty. {}
I tried many things (adding uniformsNeedsUpadte etc..) but no way to update 'time' uniform
There must be something I am missing..Could you give me some details / exemple on the way to set uniforms?  

 
Thanks !

**Comment by tentone** [comment]
The uniforms objects are created manually in the script. They are not automatically created from your GLSL code.

As im writing the shader tutorial i will add a uniform / attribute editor tool in the shader editor.
